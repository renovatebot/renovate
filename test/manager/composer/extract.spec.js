const fs = require('fs');
const {
  extractDependencies,
} = require('../../../lib/manager/composer/extract');

const requirements1 = fs.readFileSync(
  'test/_fixtures/composer/composer1.json',
  'utf8'
);

describe('lib/manager/composer/extract', () => {
  describe('extractDependencies()', () => {
    let packageFile;
    beforeEach(() => {
      packageFile = 'composer.json';
    });
    it('returns null for invalid json', async () => {
      expect(await extractDependencies('nothing here', packageFile)).toBe(null);
    });
    it('returns null for empty deps', async () => {
      expect(await extractDependencies('{}', packageFile)).toBe(null);
    });
    it('extracts dependencies with no lock file', async () => {
      const res = await extractDependencies(requirements1, packageFile);
      expect(res).toMatchSnapshot();
    });
    it('extracts dependencies with lock file', async () => {
      platform.getFile.mockReturnValueOnce('some content');
      const res = await extractDependencies(requirements1, packageFile);
      expect(res).toMatchSnapshot();
    });
  });
});
