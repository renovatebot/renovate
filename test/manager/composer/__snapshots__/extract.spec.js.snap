// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/manager/composer/extract extractDependencies() extracts dependencies with lock file 1`] = `
Object {
  "composerLock": "composer.lock",
  "deps": Array [
    Object {
      "currentValue": ">=5.3.2",
      "depName": "php",
      "depType": "require",
      "purl": "pkg:packagist/php",
      "skipReason": "unsupported",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "symfony/assetic-bundle",
      "depType": "require",
      "purl": "pkg:packagist/symfony/assetic-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "symfony/monolog-bundle",
      "depType": "require",
      "purl": "pkg:packagist/symfony/monolog-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "symfony/swiftmailer-bundle",
      "depType": "require",
      "purl": "pkg:packagist/symfony/swiftmailer-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "2.1.*",
      "depName": "symfony/symfony",
      "depType": "require",
      "purl": "pkg:packagist/symfony/symfony",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "2.2.2",
      "depName": "doctrine/common",
      "depType": "require",
      "purl": "pkg:packagist/doctrine/common",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "doctrine/doctrine-bundle",
      "depType": "require",
      "purl": "pkg:packagist/doctrine/doctrine-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "doctrine/doctrine-fixtures-bundle",
      "depType": "require",
      "purl": "pkg:packagist/doctrine/doctrine-fixtures-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "2.2.x-dev",
      "depName": "doctrine/orm",
      "depType": "require",
      "purl": "pkg:packagist/doctrine/orm",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "exercise/elastica-bundle",
      "depType": "require",
      "purl": "pkg:packagist/exercise/elastica-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "friendsofsymfony/rest-bundle",
      "depType": "require",
      "purl": "pkg:packagist/friendsofsymfony/rest-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "*",
      "depName": "friendsofsymfony/user-bundle",
      "depType": "require",
      "purl": "pkg:packagist/friendsofsymfony/user-bundle",
      "skipReason": "any-version",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "*",
      "depName": "fzaninotto/faker",
      "depType": "require",
      "purl": "pkg:packagist/fzaninotto/faker",
      "skipReason": "any-version",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "1.0.1",
      "depName": "jms/di-extra-bundle",
      "depType": "require",
      "purl": "pkg:packagist/jms/di-extra-bundle",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "*",
      "depName": "jms/payment-core-bundle",
      "depType": "require",
      "purl": "pkg:packagist/jms/payment-core-bundle",
      "skipReason": "any-version",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "1.1.0",
      "depName": "jms/security-extra-bundle",
      "depType": "require",
      "purl": "pkg:packagist/jms/security-extra-bundle",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "knplabs/knp-menu-bundle",
      "depType": "require",
      "purl": "pkg:packagist/knplabs/knp-menu-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "knplabs/knp-paginator-bundle",
      "depType": "require",
      "purl": "pkg:packagist/knplabs/knp-paginator-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "liip/imagine-bundle",
      "depType": "require",
      "purl": "pkg:packagist/liip/imagine-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "merk/dough-bundle",
      "depType": "require",
      "purl": "pkg:packagist/merk/dough-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "sensio/distribution-bundle",
      "depType": "require",
      "purl": "pkg:packagist/sensio/distribution-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "sensio/framework-extra-bundle",
      "depType": "require",
      "purl": "pkg:packagist/sensio/framework-extra-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "sensio/generator-bundle",
      "depType": "require",
      "purl": "pkg:packagist/sensio/generator-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "simplethings/entity-audit-bundle",
      "depType": "require",
      "purl": "pkg:packagist/simplethings/entity-audit-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "stof/doctrine-extensions-bundle",
      "depType": "require",
      "purl": "pkg:packagist/stof/doctrine-extensions-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "twig/extensions",
      "depType": "require",
      "purl": "pkg:packagist/twig/extensions",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "2.3.*",
      "depName": "behat/behat",
      "depType": "require-dev",
      "purl": "pkg:packagist/behat/behat",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "*",
      "depName": "behat/behat-bundle",
      "depType": "require-dev",
      "purl": "pkg:packagist/behat/behat-bundle",
      "skipReason": "any-version",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "*",
      "depName": "behat/mink-bundle",
      "depType": "require-dev",
      "purl": "pkg:packagist/behat/mink-bundle",
      "skipReason": "any-version",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "*",
      "depName": "behat/sahi-client",
      "depType": "require-dev",
      "purl": "pkg:packagist/behat/sahi-client",
      "skipReason": "any-version",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "*",
      "depName": "behat/common-contexts",
      "depType": "require-dev",
      "purl": "pkg:packagist/behat/common-contexts",
      "skipReason": "any-version",
      "versionScheme": "semverComposer",
    },
  ],
}
`;

exports[`lib/manager/composer/extract extractDependencies() extracts dependencies with no lock file 1`] = `
Object {
  "composerLock": false,
  "deps": Array [
    Object {
      "currentValue": ">=5.3.2",
      "depName": "php",
      "depType": "require",
      "purl": "pkg:packagist/php",
      "skipReason": "unsupported",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "symfony/assetic-bundle",
      "depType": "require",
      "purl": "pkg:packagist/symfony/assetic-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "symfony/monolog-bundle",
      "depType": "require",
      "purl": "pkg:packagist/symfony/monolog-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "symfony/swiftmailer-bundle",
      "depType": "require",
      "purl": "pkg:packagist/symfony/swiftmailer-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "2.1.*",
      "depName": "symfony/symfony",
      "depType": "require",
      "purl": "pkg:packagist/symfony/symfony",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "2.2.2",
      "depName": "doctrine/common",
      "depType": "require",
      "purl": "pkg:packagist/doctrine/common",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "doctrine/doctrine-bundle",
      "depType": "require",
      "purl": "pkg:packagist/doctrine/doctrine-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "doctrine/doctrine-fixtures-bundle",
      "depType": "require",
      "purl": "pkg:packagist/doctrine/doctrine-fixtures-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "2.2.x-dev",
      "depName": "doctrine/orm",
      "depType": "require",
      "purl": "pkg:packagist/doctrine/orm",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "exercise/elastica-bundle",
      "depType": "require",
      "purl": "pkg:packagist/exercise/elastica-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "friendsofsymfony/rest-bundle",
      "depType": "require",
      "purl": "pkg:packagist/friendsofsymfony/rest-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "*",
      "depName": "friendsofsymfony/user-bundle",
      "depType": "require",
      "purl": "pkg:packagist/friendsofsymfony/user-bundle",
      "skipReason": "any-version",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "*",
      "depName": "fzaninotto/faker",
      "depType": "require",
      "purl": "pkg:packagist/fzaninotto/faker",
      "skipReason": "any-version",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "1.0.1",
      "depName": "jms/di-extra-bundle",
      "depType": "require",
      "purl": "pkg:packagist/jms/di-extra-bundle",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "*",
      "depName": "jms/payment-core-bundle",
      "depType": "require",
      "purl": "pkg:packagist/jms/payment-core-bundle",
      "skipReason": "any-version",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "1.1.0",
      "depName": "jms/security-extra-bundle",
      "depType": "require",
      "purl": "pkg:packagist/jms/security-extra-bundle",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "knplabs/knp-menu-bundle",
      "depType": "require",
      "purl": "pkg:packagist/knplabs/knp-menu-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "knplabs/knp-paginator-bundle",
      "depType": "require",
      "purl": "pkg:packagist/knplabs/knp-paginator-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "liip/imagine-bundle",
      "depType": "require",
      "purl": "pkg:packagist/liip/imagine-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "merk/dough-bundle",
      "depType": "require",
      "purl": "pkg:packagist/merk/dough-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "sensio/distribution-bundle",
      "depType": "require",
      "purl": "pkg:packagist/sensio/distribution-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "sensio/framework-extra-bundle",
      "depType": "require",
      "purl": "pkg:packagist/sensio/framework-extra-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "sensio/generator-bundle",
      "depType": "require",
      "purl": "pkg:packagist/sensio/generator-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "simplethings/entity-audit-bundle",
      "depType": "require",
      "purl": "pkg:packagist/simplethings/entity-audit-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "stof/doctrine-extensions-bundle",
      "depType": "require",
      "purl": "pkg:packagist/stof/doctrine-extensions-bundle",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "dev-master",
      "depName": "twig/extensions",
      "depType": "require",
      "purl": "pkg:packagist/twig/extensions",
      "skipReason": "unsupported-constraint",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "2.3.*",
      "depName": "behat/behat",
      "depType": "require-dev",
      "purl": "pkg:packagist/behat/behat",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "*",
      "depName": "behat/behat-bundle",
      "depType": "require-dev",
      "purl": "pkg:packagist/behat/behat-bundle",
      "skipReason": "any-version",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "*",
      "depName": "behat/mink-bundle",
      "depType": "require-dev",
      "purl": "pkg:packagist/behat/mink-bundle",
      "skipReason": "any-version",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "*",
      "depName": "behat/sahi-client",
      "depType": "require-dev",
      "purl": "pkg:packagist/behat/sahi-client",
      "skipReason": "any-version",
      "versionScheme": "semverComposer",
    },
    Object {
      "currentValue": "*",
      "depName": "behat/common-contexts",
      "depType": "require-dev",
      "purl": "pkg:packagist/behat/common-contexts",
      "skipReason": "any-version",
      "versionScheme": "semverComposer",
    },
  ],
}
`;
